### ISSUE TITLE
Method createPdoInstance() in \yii\db\Connection have to guarantee $this->attributes is array

### ISSUE THREAD

**Issue by neutrinobg** [yiisoft/yii2#10352]
I think that since the PHP PDO Manual says:
public __construct ( string $dsn [, string $username [, string $password [, array $options ]]] )
for the class PDO, the method createPdoInstance() in \yii\db\Connection have to guarantee $this->attributes is array. As it is written now null value could be passed.

**Comment by klimov-paul** [comment]
No, extra type check is redundant here. If wrong type will be passed to the PDO constructor it will raise error on its own notifying developer.

**Comment by neutrinobg** [comment]
Thanks for the promt response. I am writing a custom PDO class that's why I wrote the post.
I followed the documentation and declared my constructor this way:
public function __construct($dsn, $username = null, $passwd = null, array $options = array()) 
and with correct connection configuration i got this unexpected to me error:
Argument 4 passed to oci2pdo\OciPdoAdapter::__construct() must be of the type array, null given, called in D:\nginx\html\yiidemo\advanced\vendor\yiisoft\yii2\db\Connection.php on line 579 and defined
That is why I thought it is more appropriate the value for the last parameter not to be null but empty array.
I changed my costructor and deleted the array type declaration before the last parameter.
